schedule.py

```diff
@@ -282,9 +277,15 @@ def print_results(solver, classes, professors, sections, semesters):
             for _, professor in sorted(professors.items()):
                 if solver.Value(classes[(professor.name, section.name)]) == 1:
                     if professor.prefers(section.course):
-                        print(section.name + ' assigned to ' + professor.name + ' (requested)')
+                        if section.must_offer:
+                            print(section.name + ' assigned to ' + professor.name + ' (required) (requested)')
+                        else:
+                            print(section.name + ' assigned to ' + professor.name + ' (optional) (requested)')
                     else:
-                        print(section.name + ' assigned to ' + professor.name + ' (not requested)')
+                        if section.must_offer:
+                            print(section.name + ' assigned to ' + professor.name + ' (required) (not requested)')
+                        else:
+                            print(section.name + ' assigned to ' + professor.name + ' (optional) (not requested)')
```

Comment:
I'm okay with leaving out "optional"/"required" here, which gets messy when there you require multiple sections.